source/common/tcp_proxy/upstream.cc

```diff
@@ -54,17 +56,10 @@ TcpUpstream::onDownstreamEvent(Network::ConnectionEvent event) {
 
 HttpUpstream::HttpUpstream(Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
                            const std::string& hostname)
-    : upstream_callbacks_(callbacks), response_decoder_(*this), hostname_(hostname) {}
+    : hostname_(hostname), response_decoder_(*this), upstream_callbacks_(callbacks) {}
```

Comment:
did the ordering need to change here?  If not let's revert.

Reply:
yep, this is needed because I moved `hostname_` to protected